cmd/scollector/collectors/collectors.go

```diff
@@ -266,7 +258,20 @@ func AddTS(md *opentsdb.MultiDataPoint, name string, ts int64, value interface{}
 		Value:     value,
 		Tags:      tags,
 	}
-	*md = append(*md, &d)
+	if d.Valid() {
+		*md = append(*md, &d)
+		if rate != metadata.Unknown {
+			metadata.AddMeta(name, nil, "rate", rate, false)
+		}
+		if unit != metadata.None {
+			metadata.AddMeta(name, nil, "unit", unit, false)
+		}
+		if desc != "" {
+			metadata.AddMeta(name, tags, "desc", desc, false)
+		}
+	} else {
+		slog.Errorf("Invalid datapoint received for: %s", name)
```

Comment:
Would probably be useful to print all fields of the datapoint so one can see what was wrong with it.